Documentation/authentication.md

```diff
@@ -0,0 +1,164 @@
+# Authentication Guide of etcd v3
+
+## Overview
+
+Authentication was added in etcd 2.1. In the v3 API, the authentication feature, its API and user interface, was modified slightly. This guide will help users set up basic authentication in etcd v3.
+
+## Special Users and Roles
+
+There is one special user, `root`, and there are two special roles, `root` and `guest`.
+
+### User `root`
+
+User `root` must be created before auth can be activated. It must have the `root` role and is allowed to change anything inside etcd. The idea behind the `root` user is for administrative purposes: managing other ordinal users and roles.  In addition, the root role is required for other management functionalities including adding/removing/updating a node in a cluster, defragmentation, taking a snapshot, and checking endpoint health.
```

Comment:
The `root` user, which has full access to etcd, must be created before activating authentication. The idea behind the `root` user is for administrative purposes: managing roles and ordinary users.  The `root` user must have the `root` role and is allowed to change anything inside etcd.
?

Is `ordinal` meant to be `ordinary`? I've never seen `ordinal` used this way (aside from a few comments in the etcd code).